connectors/src/connectors/zendesk/lib/sync_article.ts

```diff
@@ -19,6 +20,36 @@ import type { DataSourceConfig } from "@connectors/types/data_source_config";
 
 const turndownService = new TurndownService();
 
+/**
+ * Deletes an article from the db and the data sources.
+ */
+export async function deleteArticle(
+  connectorId: ModelId,
+  article: ZendeskFetchedArticle,
+  dataSourceConfig: DataSourceConfig,
+  loggerArgs: Record<string, string | number | null>
+): Promise<void> {
+  logger.info(
+    {
+      ...loggerArgs,
+      connectorId,
+      articleId: article.id,
+      name: article.name,
+    },
+    "[Zendesk] Deleting article."
+  );
+  await Promise.all([
+    ZendeskArticleResource.deleteByArticleId({
+      connectorId,
+      articleId: article.id,
+    }),
+    deleteFromDataSource(
+      dataSourceConfig,
+      getArticleInternalId(connectorId, article.id)
+    ),
+  ]);
+}
+
```

Comment:
note: I wanted to implement article deletion similarly to for the tickets, unfortunately the endpoint does not return entries for deleted or archived articles (nor does it provide a status for the articles).
A dedicated garbage collection will be implemented as per [#8656](https://github.com/dust-tt/dust/issues/8656)